packages/p2p-v2/src/components/Modals/NicknameModal/NicknameModal.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { ComponentType, useEffect } from 'react';
+import { Controller, useForm, useWatch } from 'react-hook-form';
+import Modal from 'react-modal';
+import { p2p } from '@deriv/api';
+import { Button } from '@deriv-com/ui/dist/components/Button';
+import { Text } from '@deriv-com/ui/dist/components/Text';
+import { useDevice } from '../../../hooks';
+import P2PUserIcon from '../../../public/ic-cashier-p2p-user.svg';
+import { Input } from '../../Input';
+import { customStyles } from '../helpers';
+import './NicknameModal.scss';
+
+const NicknameModal = () => {
+    const ReactModal = Modal as ComponentType<ReactModal['props']>;
+    const {
+        control,
+        formState: { errors },
+        getValues,
+        handleSubmit,
+    } = useForm({
+        defaultValues: {
+            nickname: '',
+        },
+        mode: 'onChange',
+    });
+
+    const { error, isError, mutate, reset } = p2p.advertiser.useCreate();
+    const { isMobile } = useDevice();
+
+    const onSubmit = () => {
+        mutate({ name: getValues('nickname') });
+        setTimeout(() => {
+            reset();
+        }, 3000);
+    };
+
+    const errorMessage = error?.error?.message || 'Can only contain letters, numbers, and special characters .-_@.';
+    const hasError = errors?.nickname?.type === 'pattern' || isError;
+    const watchNickname = useWatch({ control, defaultValue: '', name: 'nickname' });
+
+    useEffect(() => {
+        Modal.setAppElement('#v2_modal_root');
+    }, []);
+
+    return (
+        <ReactModal
+            className='p2p-v2-nickname-modal'
+            isOpen={true}
+            onRequestClose={() => {
+                // Implement return function here
+            }}
```

Comment:
is this needed?

Reply:
i had to leave a comment here cause eslint was giving an error